src/Events.Functions/Clients/EventsStorageClient.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Altinn.Common.AccessTokenClient.Services;
+using Altinn.Platform.Events.Functions.Clients.Interfaces;
+using Altinn.Platform.Events.Functions.Configuration;
+using Altinn.Platform.Events.Functions.Extensions;
+using Altinn.Platform.Events.Functions.Models;
+using Altinn.Platform.Events.Functions.Services.Interfaces;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.Platform.Events.Functions.Clients
+{
+    /// <summary>
+    /// Events Storage REST API client
+    /// </summary>
+    public class EventsStorageClient : IEventsStorageClient
```

Comment:
There is a lot of shared code between the new clients. Could we maybe share more of it by creaing an astract class that handles generateAccessToken?